plugins/network-elements/nuage-vsp/src/com/cloud/network/resource/NuageVspResource.java

```diff
@@ -170,20 +169,18 @@ public boolean configure(String name, Map<String, Object> params) throws Configu
             throw new ConfigurationException("Unable to find retry interval");
         }
 
-        _relativePath = new StringBuffer().append("https://").append(hostname).append(":").append(port).append(apiRelativePath).toString();
+        _relativePath = new StringBuffer().append("https://").append(_hostName).append(":").append(port).append(apiRelativePath).toString();
 
         String cmsUserPass = org.apache.commons.codec.binary.StringUtils.newStringUtf8(Base64.decodeBase64(cmsUserPassBase64));
         _cmsUserInfo = new String[] {CMS_USER_ENTEPRISE_NAME, cmsUser, cmsUserPass};
 
-        try {
-            loadNuageClient();
-        } catch (Exception e) {
-            throw new CloudRuntimeException("Failed to login to Nuage VSD on " + name + " as user " + cmsUser, e);
-        }
+        _nuageVspCmsId = (String)params.get("nuagevspcmsid");
+
+        loadNuageClient();
 
         try {
             login();
-        } catch (Exception e) {
+        } catch (ExecutionException | ConfigurationException e) {
             s_logger.error("Failed to login to Nuage VSD on " + name + " as user " + cmsUser + " Exception " + e.getMessage());
```

Comment:
Add the exception to the log statement to ensure the stack trace is added to the log.